openfisca_core/tracers/computation_log.py

```diff
@@ -60,12 +61,16 @@ def print_line(depth: int, node: tracers.TraceNode) -> str:
 
             return f"{indent}{node.name}<{node.period}> >> {formatted_value}"
 
+        if (depth_max is not None) and (depth > depth_max):
+            return [""]
+
         node_log = [print_line(depth, node)]
 
         children_logs = [
-            self._get_node_log(child, depth + 1, aggregate)
+            self._get_node_log(child, depth + 1, aggregate, depth_max = depth_max)
             for child
             in node.children
+            if self._get_node_log(child, depth + 1, aggregate, depth_max = depth_max) != [""]
             ]
 
         return node_log + self._flatten(children_logs)
```

Comment:
```suggestion
        return node_log + self._compact(self._flatten(children_logs))
        
    def _compact(self, list: List[Optional[str]]) -> List[str]:
        return [item for item in list if item is not None] 
```

Reply:
I had problems with types. that's why I used that ugly ` != [""]` ;-)
Since there are many good suggestions @maukoquiroga, I propose that you push all of them and I will react to them afterwards if you don't mind ?

Reply:
Ah was it about types, sorry to read that.

Reply:
These are just suggestions, you can not agree with them.

Reply:
I agree with you that it is ugly but I didn't find a proper way to avoid it so I suggested that you handle it.
BTW I will write the test ASAP.